src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { useDispatch, useSelector } from 'react-redux';
+import Modal from "../../common/modal/Modal";
+import ModalTheme from '../ledger/ModalTheme';
+import MobileActionSheet from '../../common/modal/MobileActionSheet';
+import FormButton from '../../common/FormButton';
+import { Translate } from 'react-localize-redux';
+import TwoFactorVerifyInput from './TwoFactorVerifyInput';
+import { verifyTwoFactor, clearAlert, resendTwoFactor } from '../../../actions/account';
+import { wallet } from '../../../utils/wallet';
+
+const Form = styled.form`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    width: 100%;
+`
+
+const TwoFactorVerifyModal = ({ open, onClose }) => {
+
+    const [method, setMethod] = useState();
+    const [code, setCode] = useState('');
+    const dispatch = useDispatch();
+    const account = useSelector(({ account }) => account);
+    const loading = account.actionsPending.includes('VERIFY_TWO_FACTOR');
```

Comment:
Wonder if we shouldn't actually check for specific actions. Instead we can have info in `meta` whether given action are blocking (most actions initiated by user) or not (refreshing transaction history on timer, checking account availability when typing a name, etc).

Otherwise we won't track cases involving multiple cascading actions well.